### PR TITLE
fix encoding

### PR DESCRIPTION
**Pull request by odinuv**
feat: baseurl encode encryption input
feat: optionally baseurl decode keyoperation result
refactor: split EncryptDecryptRequest into EncryptRequest and DecryptRequest

### CODE REVIEW

File: src/Base64UrlEncoder.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    public static function encode($data)
+    {
+        return rtrim(strtr(base64_encode($data), '+/', '-_'), '=');
+    }
+
+    public static function decode($data)
+    {
+        return base64_decode(strtr($data, '-_', '+/'), true);
+    }
+}
```
Comment (MiroCillik) [inline src/Base64UrlEncoder.php:19]: Ty tam aj nahradzujes nejake znaky? Lebo podobne to riesim v OAuth API ale bez roho replacu:
https://github.com/keboola/oauth-api/blob/master/src/lib/Encryption/KeyVaultEncryption.js#L28

File: tests/Base64UrlEncoderTest.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                '++++',
+                'KysrKw',
+            ],
+            [
+                '+++++',
+                'KysrKys',
+            ],
+        ];
+    }
+}
```
Comment (MiroCillik) [inline tests/Base64UrlEncoderTest.php:51]: To schvalne vyskusam v tom mojom :)

**Review by MiroCillik — APPROVED** [review]
Myslim, ze dobre. Musim tie "+"ka a podobne znaky vyskusat aj v OAuth API (node.js), ze ci to tiez musim takto replacovat.

File: src/Base64UrlEncoder.php (hunk 1 above)
Comment (odinuv) [reply src/Base64UrlEncoder.php:19]: jo, tam je rozdil v tom, ze PHP produkuje base64 a azure API ocekava base64url, coz je malinko jiny, ale jen trosku https://brockallen.com/2014/10/17/base64url-encoding/#:~:text=Base64url%20encoding%20is%20basically%20base64,a%20replacement%20for%20base64%20encoding.
takze zalezi na tom co ten toString('base64') dela, PHP podporu pro base64url nema, proto je to tam takhle

**Pull request merged**